dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,158 @@
+        when(user.getLocale()).thenReturn(Locale.getDefault());
+        when(user.isBackendUser()).thenReturn(true);
+
+        final WebResource webResource = mock(WebResource.class);
+        final InitDataObject dataObject = mock(InitDataObject.class);
+        when(dataObject.getUser()).thenReturn(user);
+
+        when(webResource.init(any(WebResource.InitBuilder.class))).thenReturn(dataObject);
+
+        //ServiceIntegrationAPI serviceIntegrationAPI = new ServiceIntegrationAPIImpl();
+
+        ServiceIntegrationHelper serviceIntegrationHelper = new ServiceIntegrationHelper();
+
+        serviceIntegrationResource = new ServiceIntegrationResource(webResource, serviceIntegrationHelper);
+    }
+
+    private InputStream createServiceDescriptorFile(final String fileName, final String serviceKey,
+            String serviceName, final String description,
+            final Map<String, Param> paramMap) throws IOException {
+        final ServiceDescriptor serviceDescriptor = new ServiceDescriptor(serviceKey, serviceName, description, "/black.png");
+
+        for (final Entry<String, Param> entry : paramMap.entrySet()) {
+            final Param param = entry.getValue();
+            serviceDescriptor.addParam(entry.getKey(),param.getValue(), param.isHidden(), param.getType(), param.getLabel(), param.getHint());
+        }
+
+        String basePath = System.getProperty("java.io.tmpdir");
+        basePath = Paths.get(basePath).normalize().toString();
+        final File file = new File(basePath,fileName);
+        ymlMapper.writeValue(file, serviceDescriptor);
+        System.out.println(file);
+        return new FileInputStream(file);
+    }
+
+    private FormDataMultiPart createFormDataMultiPart(final String fileName, final InputStream inputStream) {
+        final BodyPart filePart1 = new StreamDataBodyPart(fileName, inputStream);
+        final FormDataMultiPart formDataMultiPart = new FormDataMultiPart();
+        formDataMultiPart.bodyPart(filePart1);
+        return formDataMultiPart;
+    }
+
+    @Test
+    public void  Test_Create_service_descriptor_Then_Create_Service_Integration() throws IOException {
+        final Map<String, Param> paramMap = ImmutableMap.of(
+                "p1", Param.newParam("v1", false, Type.STRING, "label", "hint"),
+                "p2", Param.newParam("v2", false, Type.STRING, "label", "hint"),
+                "p3", Param.newParam("v3", false, Type.STRING, "label", "hint")
+        );
+        final Host host = new SiteDataGen().nextPersisted();
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        final String serviceKey = String.format("lol_%d",System.currentTimeMillis());
+        final String fileName = String.format("%s.yml",serviceKey);
+        final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, "lola",
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)